frontend/store/index.js

```diff
@@ -30,23 +30,54 @@ export const getters = {
   }
 }
 
+function handleErrorAxios (error) {
+  // this is optional and mostly for debugging on dev mode
+  if (error.response) {
+    // Request made and server responded
+    console.log(error.response.data)
+    console.log(error.response.status)
+    console.log(error.response.headers)
+  } else if (error.request) {
+    // The request was made but no response was received
+    console.log(error.request)
+  } else {
+    // Something happened in setting up the request that triggered an Error
+    console.log('Error', error.message)
+  }
+  throw error
+}
+
 export const actions = {
-  async nuxtServerInit ({ dispatch }, { req }) {
-    await dispatch('getJobsForPage', { page: 0, limit: 50 })
-    await dispatch('getContentServers')
+  async nuxtServerInit ({ dispatch }, { error }) {
+    try {
+      await dispatch('getJobsForPage', { page: 0, limit: 50 })
+      await dispatch('getContentServers')
+    } catch (e) {
+      error({ statusCode: 504, message: 'Backend is unreachable! Is the backend server up and running?' })
+    }
   },
   async getJobsForPage ({ commit }, { page, limit }) {
-    const response = await this.$axios.$get(`/api/jobs/pages/${page}?limit=${limit}`)
-    const data = []
-    response.forEach(function (item) {
-      data.push(flattenObject(item))
-    })
-    // if (data.hasOwnProperty('collection_id')) {
-    commit('REFRESH_JOBS', data)
-    // }
+    await this.$axios.$get(`/api/jobs/pages/${page}?limit=${limit}`)
+      .then((response) => {
```

Comment:
Seems strange to use callback syntax in an async function, unless you are explicitly trying to not await something. But this only makes a single request. Could this be written using async-await and a try-catch block?

Reply:
hm, good point. It's maybe strange, or not good practice, to use await when not await something. The `then()` and `catch()` blocks are promises which are called after the `$get` but since nothing is happening after await I think I mixed it them both too much up after reading [this article](https://medium.com/better-programming/should-i-use-promises-or-async-await-126ab5c98789)